### PR TITLE
Update tests to use System.Text.Json over Newtonsoft.Json

### PR DESCRIPTION
**Pull request by TobbenTM**
*Description of changes:*

This PR updates the default tests in blueprints to use System.Text.Json for the samples instead of Newtonsoft.Json.

While the Newtonsoft.Json package might work for most things, it'll fail as soon as you try testing functions with authorizer claims, since the marshalling is expecting the Claims property to be populated: https://github.com/aws/aws-lambda-dotnet/blob/baa10d4788f339bd35137ea1b5c01df0a5b82afc/Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs#L125

This is because the Claims getter is looking explicitly for a `JsonElement` claims object, it'll fail if you pass a Newtonsoft.Json deserialized payload as that will have a `JObject` type there instead: https://github.com/aws/aws-lambda-dotnet/blob/087590ce99274e16e26d37e1dfd73b0b71d1230a/Libraries/src/Amazon.Lambda.APIGatewayEvents/APIGatewayCustomAuthorizerContext.cs#L136

Additionally, I expanded the current test in `TestCallingWebAPI.cs` to validate this issue. When using Newtonsoft.Json on the `TestGetCustomAuthorizerValue` test, it'll fail with an empty `Claims` dictionary, even if it exists in the sample request.

I did not add directives to support `NETSTANDARD_2_0` since the test project as a whole does not have a configuration for that.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/S3ProxyControllerTests.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Text.Json;
```
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/S3ProxyControllerTests.cs:5]: The blueprints in VS2017 folder target .NET Core 2.1 which doesn't have System.Text.Json. I suggest leaving the 2017 blueprints alone.

File: Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.cs [hunk 2]
```diff
@@ -2,15 +2,14 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Text.Json;
```
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.cs:5]: Same comment as above. Don't change the VS 2017 blueprints.

**Review by normj — CHANGES_REQUESTED** [review]

File: Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/S3ProxyControllerTests.cs (hunk 1 above)
Comment (TobbenTM) [reply Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/S3ProxyControllerTests.cs]: Okay, reverted those changes now 👍

File: Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.cs (hunk 2 above)
Comment (TobbenTM) [reply Blueprints/BlueprintDefinitions/vs2017/AspNetCoreWebAPI/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.cs]: Reverted.

**Review by philasmar — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Comment by normj** [comment]
Thank you for the PR

**Pull request merged**